flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java

```diff
@@ -130,15 +135,15 @@ public void testUpsertAndQuery() {
 
     try {
       sql("INSERT INTO %s VALUES " +
-          "(1, 'a', TO_DATE('2022-03-01'))," +
-          "(2, 'b', TO_DATE('2022-03-01'))," +
-          "(1, 'b', TO_DATE('2022-03-01'))",
+          "(1, 'a', DATE '2022-03-01')," +
+          "(2, 'b', DATE '2022-03-01')," +
+          "(1, 'b', DATE '2022-03-01')",
           tableName);
 
       sql("INSERT INTO %s VALUES " +
-          "(4, 'a', TO_DATE('2022-03-02'))," +
-          "(5, 'b', TO_DATE('2022-03-02'))," +
-          "(1, 'b', TO_DATE('2022-03-02'))",
+          "(4, 'a', DATE '2022-03-02')," +
+          "(5, 'b', DATE '2022-03-02')," +
+          "(1, 'b', DATE '2022-03-02')",
```

Comment:
Can we addd a comment about the usage of `DATE` vs `TO_DATE`, given that we're relying on `DATE` to affect the generated job graph differently than `TO_DATE?

A short comment and then directing to this comment might be sufficent: https://github.com/apache/iceberg/issues/4515#issuecomment-1094462606

Reply:
I think the commit message is a more proper place to explain it. 
I'll split this change into two parts, and describe the reason in the commit message of the first. 
WDYT @kbendick ?

Reply:
+1. As well as a proper description in the PR comment (which can be the same). As those are the two most commonly searched places when reviewing older things.

Thanks again for the great work.